### PR TITLE
[IMPROVE] Rewrite remove room invite modal

### PR DESCRIPTION
**Pull request by dougfabris**
### before
image

### after
image

## Steps to test or reproduce

- First create a invite link: go to some room, click on members and invite link button
- Go to admin > Invites
- Click on the "X" button in some invite row to see the changes

### CODE REVIEW

File: client/views/admin/invites/InviteRow.tsx [hunk 1]
```diff
@@ -1,25 +1,43 @@
...
+	const daysToExpire = ({
+		expires,
+		days,
+	}: {
+		expires: IInvite['expires'];
+		days: IInvite['days'];
+	}): string => {
 		if (days > 0) {
-			if (expires < Date.now()) {
+			if (expires && expires.getTime() < new Date().getTime()) {
```
Comment (ujorgeleite) [inline client/views/admin/invites/InviteRow.tsx:40]: maybe decoupling a method isExpired here. Is possible to get days null or undefined?

File: client/views/admin/invites/InviteRow.tsx [hunk 2]
```diff
@@ -29,7 +47,13 @@ function InviteRow({ _id, createdAt, expires, days, uses, maxUses, onRemove }) {
...
-	const maxUsesLeft = ({ maxUses, uses }) => {
+	const maxUsesLeft = ({
+		maxUses,
+		uses,
+	}: {
+		maxUses: IInvite['maxUses'];
+		uses: IInvite['uses'];
+	}): number | string => {
 		if (maxUses > 0) {
 			if (uses >= maxUses) {
```
Comment (ujorgeleite) [inline client/views/admin/invites/InviteRow.tsx:58]: Here we can decouple a method for excededMaximun or something like this I think

**Review by ujorgeleite — CHANGES_REQUESTED** [review]

File: client/views/admin/invites/InviteRow.tsx (hunk 1 above)
Comment (dougfabris) [reply client/views/admin/invites/InviteRow.tsx:40]: It's possible to get null

File: client/views/admin/invites/InviteRow.tsx (hunk 2 above)
Comment (dougfabris) [reply client/views/admin/invites/InviteRow.tsx:58]: I don't think it deserves a method to check this simple thing!

File: client/views/admin/invites/InviteRow.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+			return maxUses - uses;
+		}
+
+		return t('Unlimited');
+	};
+
+	const handleRemoveButtonClick = async (event: MouseEvent<HTMLElement>): Promise<void> => {
+		event.stopPropagation();
+
+		imperativeModal.open({
```
Comment (gabriellsh) [inline client/views/admin/invites/InviteRow.tsx:79]: Please, use the modal context istead of this.

**Review by gabriellsh — CHANGES_REQUESTED** [review]

File: client/views/admin/invites/InviteRow.tsx (hunk 2 above)
Comment (ujorgeleite) [reply client/views/admin/invites/InviteRow.tsx:67]: When someone goes offer manutention on this code, maybe will be better if can read what are you checking directly and not to be necessary to interpret the two-block of validation, that was my motivation to suggest this for you, but I'll approve the review.

**Review by ujorgeleite — DISMISSED** [review]

**Review by gabriellsh — APPROVED** [review]

**Review by gabriellsh — APPROVED** [review]

**Pull request merged**